044-cluster-wide-volume-usage-quota-management.md

```diff
@@ -0,0 +1,295 @@
+for producers configured with `acks >= 1` as the replication from partition leader to follower maybe delayed due to the 
+throttle.
+
+Currently, the kafka-quotas-plugin considers the total quantity of storage and how much of that is used (
+see [issue#2](https://github.com/strimzi/kafka-quotas-plugin/issues/2)) when considering whether to apply throttling to
+clients. This is problematic with respect to handling disk failure for Just a Bunch Of Disks (JBOD) deployments [(KIP-112)](https://cwiki.apache.org/confluence/display/KAFKA/KIP-112%3A+Handle+disk+failure+for+JBOD) as the
+broker will take partitions offline when the volume they are stored on runs out of space. Which in the case of unbalanced usage
+between volumes can lead to a volume running out of storage without throttling being applied.
+
+Throttling down message production on all broker nodes will protect the cluster from
+running out of disk due to replication.
+
+## Motivation
+
+Users need better protection against running out of disk on any of their nodes as it can degrade the service
+in unpredictable ways (corrupted segment logs) which could impede the interventions required to recover.
+
+## Proposal
+
+1. Extend the kafka-quotas-plugin so that we can choose to observe the disk usage 
+of all brokers in the cluster.
+2. [Deprecate](#compatibility) the existing local disk observations.
+3. Add new limit types, so that it is explicit what is being limited and better support heterogeneous disks
+   1. throttle if used bytes exceeds threshold on any volume
+   2. throttle if available bytes less than threshold on any volume
+   3. throttle if available ratio less than threshold on any volume
+4. Introduce extension points in the quotas-plugin to support pluggable sources of quotas and throttle factors
+5. Introduce a fallback throttle factor in case we cannot retrieve the cluster state
+6. strimzi-kafka-operator becomes responsible for configuring the admin client connection properties of the plugin
+
+So every broker will make its own independent throttling decision based on knowledge of the volumes on all active broker nodes.
+The brokers should all operate on a similar view of the cluster state and make a deterministic decision. If a broker detects that
+any volume in the cluster is becoming too full it will throttle production of messages. The kafka quota API isn't rich 
+enough to do anything smarter about only throttling writes to the brokers running out of space, so we fence the whole cluster.
+
+### Caveat - KRaft Disk Usage
+
+This proposal will only help prevent running out-of-disk caused by topic data. It will not prevent disks being exhausted
+due to writes to the upcoming KRaft metadata log.
+
+Currently, when using separate controller-only nodes*, those nodes are not described by `Admin#describeCluster` and we cannot use 
+`Admin#describeLogDirs` against the controllers. So the disk usage of the volume could be invisible.
```

Comment:
It's out of scope for this proposal, but I think there's a good argument for always using a separate volume for `__cluster_metadata` logs when in combined mode. It's difficult to reason over all the ways kube PVs might be implemented, but having a separate volume means there's a better chance that I/O for normal topic load won't interfere with metadata log appends.